src/System.IO.FileSystem/src/System/IO/Win32FileSystem.cs

```diff
@@ -421,180 +421,150 @@ private static SafeFileHandle OpenHandle(string fullPath, bool asDirectory)
-                                }
-                                catch (Exception e)
-                                {
-                                    if (ex == null)
-                                        ex = e;
-                                }
+                                exception = Win32Marshal.GetExceptionForWin32Error(errorCode, fileName);
                             }
-                            else
-                            {
-                                // Check to see if this is a mount point, and
-                                // unmount it.
-                                if (data.dwReserved0 == Interop.Kernel32.IOReparseOptions.IO_REPARSE_TAG_MOUNT_POINT)
-                                {
-                                    // Use full path plus a trailing '\'
-                                    string mountPoint = Path.Combine(fullPath, data.cFileName + PathHelpers.DirectorySeparatorCharAsString);
-                                    if (!Interop.Kernel32.DeleteVolumeMountPoint(mountPoint))
-                                    {
-                                         errorCode = Marshal.GetLastWin32Error();
-                                    
-                                        if (errorCode != Interop.Errors.ERROR_SUCCESS && 
-                                            errorCode != Interop.Errors.ERROR_PATH_NOT_FOUND)
-                                        {
-                                            try
-                                            {
-                                                throw Win32Marshal.GetExceptionForWin32Error(errorCode, data.cFileName);
-                                            }
-                                            catch (Exception e)
-                                            {
-                                                if (ex == null)
-                                                    ex = e;
-                                            }
-                                        }
-                                    }
-                                }
+                        }
+                    }
+                    else
+                    {
+                        // Directory, skip ".", "..".
+                        if (findData.cFileName.Equals(".") || findData.cFileName.Equals(".."))
+                            continue;
 
-                                // RemoveDirectory on a symbolic link will
-                                // remove the link itself.
-                                string reparsePoint = Path.Combine(fullPath, data.cFileName);
-                                r = Interop.Kernel32.RemoveDirectory(reparsePoint);
-                                if (!r)
+                        string fileName = findData.cFileName.Value;
+                        if ((findData.dwFileAttributes & (int)FileAttributes.ReparsePoint) == 0)
```

Comment:
They are using different data structures, so I couldn't find a decent way to fold any of this.